src/nvim/api/private/helpers.c

```diff
@@ -397,13 +397,13 @@ bool object_to_vim(Object obj, typval_T *tv, Error *err)
 
   switch (obj.type) {
     case kObjectTypeNil:
-      tv->v_type = VAR_NUMBER;
-      tv->vval.v_number = 0;
+      tv->v_type = VAR_SPECIAL;
+      tv->vval.v_number = kSpecialVarNull;
```

Comment:
Is `tv->vval.v_number` the right place to save a special value? In new line 655, special values are read from `...->vval.v_speical`.